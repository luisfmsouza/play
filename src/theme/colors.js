export const skyLighter = "#F9FAFB";
export const skyLight = "#F4F6F8";
export const sky = "#DFE3E8";
export const skyDark = "#C4CDD5";
export const inkLightest = "#919EAB";
export const inkLighter = "#637381";
export const inkLight = "#454F5B";
export const ink = "#212B36";
