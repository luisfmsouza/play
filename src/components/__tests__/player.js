import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { ThemeProvider } from "styled-components";

import * as colors from "../../theme/colors";

import Player from "../player";

describe("Player component", () => {
  const build = state => {
    const store = {
      getState: () => state,
      dispatch: jest.fn(),
      subscribe: jest.fn()
    };

    return renderer
      .create(
        <Provider store={store}>
          <ThemeProvider theme={{ colors }}>
            <Player />
          </ThemeProvider>
        </Provider>
      )
      .toJSON();
  };

  describe("when there is playlists and videos", () => {
    it("renders default component", () => {
      const state = {
        playlists: [
          [
            {
              artist: "Artist 1",
              title: "Title 1",
              videoId: "MeSu5_Do8P8"
            },
            {
              artist: "Artist 2",
              title: "Title 2",
              videoId: "249i5jNRClU"
            }
          ],
          [],
          [
            {
              artist: "Artist 3",
              title: "Title 3",
              videoId: "qSeaR7Dqvdc"
            },
            {
              artist: "Artist 4",
              title: "Title 4",
              videoId: "Dw5orx_47uY"
            },
            {
              artist: "Artist 5",
              title: "Title 5",
              videoId: "sNFndSeEt0w"
            }
          ],
          [
            {
              artist: "Artist 6",
              title: "Title 6",
              videoId: "ykHiSn1K1E"
            }
          ]
        ],
        currentPlaylist: 0,
        currentVideo: 1,
        form: { artist: "", title: "", videoId: "" },
        showForm: false
      };

      const expected = build(state);

      expect(expected).toMatchSnapshot();
    });
  });

  describe("when does not have playlists nor videos", () => {
    it("renders add playlist button component", () => {
      const state = {
        playlists: null,
        currentPlaylist: null,
        currentVideo: null,
        form: { artist: "", title: "", videoId: "" },
        showForm: false
      };

      const expected = build(state);

      expect(expected).toMatchSnapshot();
    });
  });

  describe("when does not have video on selected playlist", () => {
    it("renders add video form component", () => {
      const state = {
        playlists: [[]],
        currentPlaylist: 0,
        currentVideo: null,
        form: { artist: "", title: "", videoId: "" },
        showForm: true
      };

      const expected = build(state);

      expect(expected).toMatchSnapshot();
    });
  });
});
