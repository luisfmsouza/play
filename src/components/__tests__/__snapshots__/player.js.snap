// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player component when does not have playlists nor videos renders add playlist button component 1`] = `
<button
  className="sc-kAzzGY lmUEGD"
  onClick={[Function]}
>
  + Add your first playlist
</button>
`;

exports[`Player component when does not have video on selected playlist renders add video form component 1`] = `
<div
  className="sc-bdVaJa hxIwds"
>
  <form
    onSubmit={[Function]}
  >
    <label
      className="sc-bwzfXH dnbGwd"
      htmlFor="artist"
    >
      Artist:
    </label>
    <input
      className="sc-htpNat gBBCDj"
      id="artist"
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
    <label
      className="sc-bwzfXH dnbGwd"
      htmlFor="title"
    >
      Title:
    </label>
    <input
      className="sc-htpNat gBBCDj"
      id="title"
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
    <label
      className="sc-bwzfXH dnbGwd"
      htmlFor="url"
    >
      URL:
    </label>
    <input
      className="sc-htpNat gBBCDj"
      id="url"
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
    <button
      className="sc-bxivhb dJVtKI"
      type="submit"
    >
      Add video
    </button>
  </form>
</div>
`;

exports[`Player component when there is playlists and videos renders default component 1`] = `
<div
  className="sc-ifAKCX fKCMBT"
>
  <div
    className="sc-EHOje bUWMzt"
  >
    <iframe
      allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
      allowFullScreen={true}
      frameBorder="0"
      height="315"
      src="https://www.youtube.com/embed/249i5jNRClU"
      title="Title 2"
      width="560"
    />
  </div>
  <div
    className="sc-jzJRlG gCenpR"
  >
    <h1
      className="sc-gZMcBi eFUkCr"
    >
      Title 2
    </h1>
    <h2
      className="sc-gqjmRU dniKdl"
    >
      Artist 2
    </h2>
  </div>
  <div
    className="sc-bZQynM fLiCiD"
  >
    <h3
      className="sc-gzVnrw gwisOI"
    >
      Videos
    </h3>
    <button
      className="sc-htoDjs bFFuJY"
      onClick={[Function]}
    >
      <div
        className="sc-dnqmqq hOKQLU"
      >
        <img
          className="sc-iwsKbI hsExMc"
          src="https://placekitten.com/100/56"
        />
        <p
          className="sc-VigVT dMUWxt"
        >
          Title 1
        </p>
        <p
          className="sc-jTzLTM ijUaSo"
        >
          Artist 1
        </p>
      </div>
    </button>
    <button
      className="sc-htoDjs bFFuJY"
      onClick={[Function]}
    >
      <div
        className="sc-dnqmqq gDPkJq"
      >
        <img
          className="sc-iwsKbI hsExMc"
          src="https://placekitten.com/100/56"
        />
        <p
          className="sc-VigVT dMUWxt"
        >
          Title 2
        </p>
        <p
          className="sc-jTzLTM ijUaSo"
        >
          Artist 2
        </p>
      </div>
    </button>
  </div>
  <div
    className="sc-cSHVUG fwLHbJ"
  >
    <button
      className="sc-fjdhpX jptjDL"
      onClick={[Function]}
    >
      + add new video
    </button>
  </div>
</div>
`;
